lib/iris/tests/unit/cube/test_CubeAttrsDict.py

```diff
@@ -0,0 +1,323 @@
+        # An empty CubeAttrsDict is "equal" to various empty containers
+        attrs = CubeAttrsDict()
+        assert attrs == {}
+        assert attrs == []
+        assert attrs == ()
+
+
+class TestDictOrderBehaviour:
+    def test_ordering(self):
+        attrs = CubeAttrsDict({"a": 1, "b": 2})
+        assert list(attrs.keys()) == ["a", "b"]
+        # Remove, then reinstate 'a' : it will go to the back
+        del attrs["a"]
+        attrs["a"] = 1
+        assert list(attrs.keys()) == ["b", "a"]
+
+    def test_globals_locals_ordering(self):
+        # create attrs with a global attribute set *before* a local one ..
+        attrs = CubeAttrsDict()
+        attrs.globals.update(dict(a=1, m=3))
+        attrs.locals.update(dict(f=7, z=4))
+        # .. and check key order of combined attrs
+        assert list(attrs.keys()) == ["a", "m", "f", "z"]
+        # create the "same" thing with locals before globals, *and* different key order
+        attrs = CubeAttrsDict()
+        attrs.locals.update(dict(z=4, f=7))
+        attrs.globals.update(dict(m=3, a=1))
+        # .. this shows that the result is not affected either by alphabetical key
+        # order, or the order of adding locals/globals
+        # I.E. result is globals-in-create-order, then locals-in-create-order
+        assert list(attrs.keys()) == ["m", "a", "z", "f"]
+
+
+class TestSettingBehaviours:
+    def test_add_localtype(self):
+        attrs = CubeAttrsDict()
+        # Any attribute not recognised as global should go into 'locals'
+        attrs["z"] = 3
+        check_content(attrs, locals={"z": 3})
+
+    @pytest.mark.parametrize("attrname", _CF_GLOBAL_ATTRS)
+    def test_add_globaltype(self, attrname):
+        # These specific attributes are recognised as belonging in 'globals'
+        attrs = CubeAttrsDict()
+        attrs[attrname] = "this"
+        check_content(attrs, globals={attrname: "this"})
+
+    def test_overwrite_local(self):
+        attrs = CubeAttrsDict(locals={"a": 1})
+        attrs["a"] = "this"
+        check_content(attrs, locals={"a": "this"})
+
+    def test_overwrite_global(self):
+        attrs = CubeAttrsDict(globals={"a": 1})
+        # Since a global attr exists, it will write that instead of creating a local
+        attrs["a"] = "this"
+        check_content(attrs, globals={"a": "this"})
+
+    @pytest.mark.parametrize("global_attrname", _CF_GLOBAL_ATTRS)
+    def test_overwrite_forced_local(self, global_attrname):
+        attrs = CubeAttrsDict(locals={global_attrname: 1})
+        # The attr *remains* local, even though it would be created global by default
+        attrs[global_attrname] = 2
+        check_content(attrs, locals={global_attrname: 2})
+
+    def test_overwrite_forced_global(self):
+        attrs = CubeAttrsDict(globals={"data": 1})
+        # The attr remains local, even though it would be created local by default
+        attrs["data"] = 2
+        check_content(attrs, globals={"data": 2})
```

Comment:
This seems like a repeat of `test_overwrite_global`?

Reply:
> a repeat of `test_overwrite_global`

Hmm. Yes I have to agree, because that is _also_ using a "forced" global rather than a "natural" one.
Rather than remove this test, I instead changed `test_overwrite_global` that one to test all recognised global names.
[a1248e859](https://github.com/SciTools/iris/pull/5040/commits/53b05d3b53d5b1644034f6746bd25ec724019ae3)